sklearn/linear_model/_base.py

```diff
@@ -668,7 +679,7 @@ def rmatvec(b):
             )
 
             if y.ndim < 2:
-                self.coef_ = lsqr(X_centered, y)[0]
+                self.coef_ = lsqr(X_centered, y, atol=self.tol, btol=self.tol)[0]
             else:
                 # sparse_lstsq cannot handle y with shape (M, K)
                 outs = Parallel(n_jobs=n_jobs_)(
```

Comment:
The subsequent call to lsqr should also be udpated to pass `self.tol` as `atol` and `btol`.